lib/vdsm/storage/storageServer.py

```diff
@@ -382,14 +382,24 @@ def _get_backup_servers_option(self):
         return "backup-volfile-servers=" + ":".join(servers)
 
     def _get_gluster_volinfo(self):
+        superVdsmProxy = supervdsm.getProxy()
         try:
-            superVdsmProxy = supervdsm.getProxy()
             volinfo = superVdsmProxy.glusterVolumeInfo(self._volname,
                                                        self._volfileserver)
             return volinfo[self._volname]
-        except ge.GlusterCmdExecFailedException as e:
+        except ge.GlusterException as e:
             log.warning("Failed to get volume info: %s", e)
-            return {}
+            try:
+                if "backup-volfile-servers" in self._options:
+                    log.warning(
+                        "Trying to get volume info through backup servers: %s",
+                        self._options)
+                    volinfo = superVdsmProxy.glusterVolumeInfo(self._volname)
+                    return volinfo[self._volname]
```

Comment:
Do we always have "backup-volfile-servers" in options? If we don't this fix will not help in the cases we don't. If we are running on a hyperconverge system we have a local gluster server regardless of having "backup-volfile-servers" options.

Regardless, there is no reason to put the if inside the try block.

Warning before trying "backup-volfile-servers" is not needed, this can be info level log.

I would always retry, and add a comment explaining why we retry. Something like:

```python
    try:
        volinfo = superVdsmProxy.glusterVolumeInfo(self._volname, self._volfileserver)
        return volinfo[self._volname] 
    except ge.GlusterException as e:
        # The remote host may be down. If we are running on a hyperconverged system
        # the gluster client can use one of the other gluster servers.
        log.info("Failed to get volume info from remote server: %s",
                 self._volfile_server)
        try:
            volinfo = superVdsmProxy.glusterVolumeInfo(self._volname)
            return volinfo[self._volname] 
        except ge.GlusterException as e
            log.warning("Failed to get volume info using local server: %s", e)

    return {}
```